src/ImageSharp/Processing/Processors/Normalization/AdaptiveHistEqualizationProcessor.cs

```diff
@@ -0,0 +1,230 @@
+                                int cdfMin = this.ClipHistogramEnabled ? this.CalculateCdf(cdf, histogramCopy, maxHistIdx) : this.CalculateCdf(cdf, histogram, maxHistIdx);
+                                float numberOfPixelsMinusCdfMin = pixelsInGrid - cdfMin;
+
+                                // Map the current pixel to the new equalized value
+                                int luminance = this.GetLuminance(source[x, y], this.LuminanceLevels);
+                                float luminanceEqualized = cdf[luminance] / numberOfPixelsMinusCdfMin;
+                                targetPixels[x, y].PackFromVector4(new Vector4(luminanceEqualized));
+
+                                // Remove top most row from the histogram, mirroring rows which exceeds the borders.
+                                Span<TPixel> rowSpan = this.GetPixelRow(source, x - halfGridSize, y - halfGridSize, this.GridSize);
+                                maxHistIdx = this.RemovePixelsFromHistogram(rowSpan, histogram, this.LuminanceLevels, maxHistIdx);
+
+                                // Add new bottom row to the histogram, mirroring rows which exceeds the borders.
+                                rowSpan = this.GetPixelRow(source, x - halfGridSize, y + halfGridSize, this.GridSize);
+                                int maxIdx = this.AddPixelsTooHistogram(rowSpan, histogram, this.LuminanceLevels);
+                                if (maxIdx > maxHistIdx)
+                                {
+                                    maxHistIdx = maxIdx;
+                                }
+                            }
+                        }
+                    });
+
+                Buffer2D<TPixel>.SwapOrCopyContent(source.PixelBuffer, targetPixels);
+            }
+        }
+
+        /// <summary>
+        /// Get the a pixel row at a given position with a length of the grid size. Mirrors pixels which exceeds the edges.
+        /// </summary>
+        /// <param name="source">The source image.</param>
+        /// <param name="x">The x position.</param>
+        /// <param name="y">The y position.</param>
+        /// <param name="gridSize">The grid size.</param>
+        /// <returns>A pixel row of the length of the grid size.</returns>
+        private Span<TPixel> GetPixelRow(ImageFrame<TPixel> source, int x, int y, int gridSize)
+        {
+            if (y < 0)
+            {
+                y = Math.Abs(y);
+            }
+            else if (y >= source.Height)
+            {
+                int diff = y - source.Height;
+                y = source.Height - diff - 1;
+            }
+
+            // Special cases for the left and the right border where GetPixelRowSpan can not be used
+            if (x < 0)
+            {
+                var rowPixels = new TPixel[gridSize];
+                int idx = 0;
+                for (int dx = x; dx < x + gridSize; dx++)
+                {
+                    rowPixels[idx] = source[Math.Abs(dx), y];
```

Comment:
I feel like we can possibly do some offsetting and move the `Abs` out of the loop here.